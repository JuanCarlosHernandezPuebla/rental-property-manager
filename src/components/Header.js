import React, {Component} from 'react';
import { Link } from 'react-router-dom';

class Header extends Component {

    render() {
        return (
            <header>
                <nav className="navbar navbar-expand-lg">
                    <div className="navbar-logo">
                        <a className="navbar-brand navbar-left" href="/home">
                            <img className="img-fluid" src="/images/rpm_logo.png" width="25%" alt="logo" />
                        </a>
                    </div>

                    <ul className="navbar-nav mr-auto">
                        <li className="nav-item active">
                            <Link className="nav-link" to='/home'>Home</Link>
                        </li>
                        <li className="nav-item">
                            <Link className="nav-link" to='/features'>Features</Link>
                        </li>
                        <li className="nav-item dropdown">
                            <a className="nav-link dropdown-toggle" href="#" id="navbarDropdown" role="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                                Resources
                                </a>
                            <div className="dropdown-menu" aria-labelledby="navbarDropdown">
                                <a className="dropdown-item" href="#">Resource #1</a>
                                <a className="dropdown-item" href="#">Resource #2</a>
                                <div className="dropdown-divider"></div>
                                <a className="dropdown-item" href="#">Sign Up</a>
                            </div>
                        </li>
                        <li className="nav-item">
                            <Link className="nav-link" to="/help">Help</Link>
                        </li>
                    </ul>
                    <ul className="navbar-nav navbar-right">
                        <li className="bold">
                            <Link className="nav-link" to="/login">Login</Link>
                        </li>
                    </ul>
                </nav>
            </header>
        );
    }

}

export default Header;