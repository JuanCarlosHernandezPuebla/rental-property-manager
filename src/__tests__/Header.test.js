import React from 'react';
import { shallow } from 'enzyme';
import Header from '../components/Header';
import renderToJson from 'enzyme-to-json';

test('Header render test', () => {
    const component = shallow(<Header/>);
    expect(renderToJson(component)).toMatchSnapshot();
});