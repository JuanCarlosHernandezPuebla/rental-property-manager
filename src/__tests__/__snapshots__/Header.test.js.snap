// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render test 1`] = `
<header>
  <nav
    className="navbar navbar-expand-lg"
  >
    <div
      className="navbar-logo"
    >
      <a
        className="navbar-brand navbar-left"
        href="/home"
      >
        <img
          alt="logo"
          className="img-fluid"
          height="122"
          src="/images/rpm_logo.png"
          width="122"
        />
      </a>
    </div>
    <ul
      className="navbar-nav mr-auto"
    >
      <li
        className="nav-item active"
      >
        <Link
          className="nav-link"
          replace={false}
          to="/home"
        >
          <FontAwesomeIcon
            border={false}
            className="mr-2"
            fixedWidth={false}
            flip={null}
            icon="home"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
          Home
        </Link>
      </li>
      <li
        className="nav-item"
      >
        <Link
          className="nav-link"
          replace={false}
          to="/features"
        >
          <FontAwesomeIcon
            border={false}
            className="mr-2"
            fixedWidth={false}
            flip={null}
            icon="star"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
          Features
        </Link>
      </li>
      <li
        className="nav-item dropdown"
      >
        <a
          aria-expanded="false"
          aria-haspopup="true"
          className="nav-link dropdown-toggle"
          data-toggle="dropdown"
          href="#"
          id="navbar_dropdown"
          role="button"
        >
          <FontAwesomeIcon
            border={false}
            className="mr-2"
            fixedWidth={false}
            flip={null}
            icon="book"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
          Resources
        </a>
        <div
          aria-labelledby="navbarDropdown"
          className="dropdown-menu"
        >
          <a
            className="dropdown-item"
            href="#"
          >
            Resource #1
          </a>
          <a
            className="dropdown-item"
            href="#"
          >
            Resource #2
          </a>
          <div
            className="dropdown-divider"
          />
          <Link
            className="nav-link"
            replace={false}
            to="/create"
          >
            Create Account
          </Link>
        </div>
      </li>
      <li
        className="nav-item"
      >
        <Link
          className="nav-link"
          replace={false}
          to="/help"
        >
          <FontAwesomeIcon
            border={false}
            className="mr-2"
            fixedWidth={false}
            flip={null}
            icon="question-circle"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
          Help
        </Link>
      </li>
    </ul>
    <ul
      className="navbar-nav navbar-right"
    >
      <li
        className="bold"
      >
        <Link
          className="nav-link"
          replace={false}
          to="/login"
        >
          Log in
          <FontAwesomeIcon
            border={false}
            className="ml-2"
            fixedWidth={false}
            flip={null}
            icon="sign-in-alt"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
        </Link>
      </li>
    </ul>
  </nav>
</header>
`;
